lib/eql.js

```diff
@@ -62,13 +62,14 @@ function deepEqual(a, b, m) {
     return bufferEqual(a, b);
   } else if ('arguments' === type(a)) {
     return argumentsEqual(a, b, m);
+  } else if ('array' === type(a)) {
+    return arrayEqual(a, b, m);
   } else if (!typeEqual(a, b)) {
     return false;
-  } else if (('object' !== type(a) && 'object' !== type(b))
-  && ('array' !== type(a) && 'array' !== type(b))) {
-    return sameValue(a, b);
-  } else {
+  } else if ('object' === type(a)) {
     return objectEqual(a, b, m);
+  } else {
+    return sameValue(a, b);
```

Comment:
Line 72 can just be `return false` because if we make it past line 55 then we already know that `sameValue(a, b)` isn't true.